File: 2572-append-characters-to-string-to-make-subsequence/2572-append-characters-to-string-to-make-subsequence.py
class Solution:
    def appendCharacters(self, s: str, t: str) -> int:
        s_length, t_length = len(s), len(t)
        s_index, t_index = 0, 0

        while s_index < s_length and t_index < t_length:
            if s[s_index] == t[t_index]:
                t_index += 1
            s_index += 1
        
        return t_length - t_index

File: 2572-append-characters-to-string-to-make-subsequence/2572-append-characters-to-string-to-make-subsequence.java
class Solution {
    public int appendCharacters(String s, String t) {
        int sIndex = 0, tIndex = 0;
        for(int i = 0; sIndex < s.length() && tIndex < t.length(); i++){
            if(s.charAt(sIndex) == t.charAt(tIndex)){
                sIndex++;
                tIndex++;
            } else {
                sIndex++;
            }
        }
        return t.length() - tIndex;
    }
}